handlers/handler.go

```diff
@@ -90,23 +94,54 @@ func (env *Env) PostUserHandler(w http.ResponseWriter, r *http.Request) {
 
 	reqUser.ID = userID
 	reqUser.Password = ""
-	location := fmt.Sprintf("%s/%d", r.URL.Path, userID)
+	location := fmt.Sprintf("%s/self", r.URL.Path)
 	w.Header().Add("Location", location)
 	w.Header().Add("Content-Type", "application/json")
 	w.WriteHeader(http.StatusCreated)
 	json.NewEncoder(w).Encode(reqUser)
 }
 
+// PatchUserHandler updates specific columns of given user
+func (env *Env) PatchUserHandler(w http.ResponseWriter, r *http.Request) {
+	userId, err := getUserID(r)
+	if err != nil {
+		errMsg := "Invalid user ID"
+		log.Println(errMsg + ": " + err.Error())
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+	reqUser := &models.User{}
+	if err := parseJSON(w, r.Body, reqUser); err != nil {
+		return
+	}
+	reqUser.ID = userId
+	if reqUser.Name == "" && reqUser.Password == "" && reqUser.AvatarURL == "" {
+		w.Header().Add("Content-Type", "application/json")
+		json.NewEncoder(w).Encode(reqUser)
+		return
+	}
+
+	retUser, err := env.DB.UpdateUser(reqUser)
+	if err != nil {
+		mySQLErr, ok := err.(*mysql.MySQLError)
+		println("code = " + string(mySQLErr.Number))
+		if ok && mySQLErr.Number == 1065 {
+			errMsg := fmt.Sprintf("User with email %s was not found", reqUser.Email)
+			log.Println(errMsg)
+			http.Error(w, errMsg, http.StatusNotFound)
+		} else {
+			internalServerError(w, err)
+		}
+		return
+	}
+
+	w.Header().Add("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(retUser)
```

Comment:
Don't return the user's ID or new password.
```suggestion
	retUser.ID = int64(0)
	retUser.Password = ""
	json.NewEncoder(w).Encode(retUser)
```